packages/expo-gl/android/src/main/java/expo/modules/gl/thread/GLSharedThread.java

```diff
@@ -0,0 +1,93 @@
+package expo.modules.gl.thread;
+
+import android.opengl.EGL14;
+import android.opengl.GLUtils;
+
+import javax.microedition.khronos.egl.EGL10;
+import javax.microedition.khronos.egl.EGLConfig;
+import javax.microedition.khronos.egl.EGLContext;
+import javax.microedition.khronos.egl.EGLDisplay;
+import javax.microedition.khronos.egl.EGLSurface;
+
+public class GLSharedThread extends GLThreadBase {
+  private static final int EGL_CONTEXT_CLIENT_VERSION = 0x3098;
+  private EGLSurface mEGLSurface;
+  private EGLContext mEGLContext;
+  private EGLContext mParentEGLContext;
+  private EGLDisplay mEGLDisplay;
+  private EGLConfig mEGLConfig;
+
+  GLSharedThread(EGLContext parentEGLContext, EGLDisplay parentEGLDisplay, EGLConfig parentEGLConfig) {
+    super();
+    mParentEGLContext = parentEGLContext;
+    mEGLDisplay = parentEGLDisplay;
+    mEGLConfig = parentEGLConfig;
+  }
+
+  @Override
+  void initEGL() {
+    mEGL = (EGL10) EGLContext.getEGL();
+
+    mEGLSurface = createEGLSurface(mEGLDisplay, mEGLConfig);
+    if (mEGLSurface == null || mEGLSurface == EGL10.EGL_NO_SURFACE) {
```

Comment:
Is it possible that `mEGLSurface` would be `null` here? I'd recommend to use `EGL_NO_*` flags in all those functions that create EGL objects.